sources/packages/backend/libs/integrations/src/institution-integration/ece-integration/ece-file.service.ts

```diff
@@ -0,0 +1,161 @@
+
+@Injectable()
+export class ECEFileService {
+  institutionIntegrationConfig: InstitutionIntegrationConfig;
+  constructor(
+    config: ConfigService,
+    private readonly eceIntegrationService: ECEIntegrationService,
+    private readonly disbursementScheduleService: DisbursementScheduleService,
+  ) {
+    this.institutionIntegrationConfig = config.institutionIntegration;
+  }
+
+  /**
+   * 1. Fetches the applications that have an eligible COE request waiting.
+   * 2. Create the Request content for the ECE request file by populating the
+   * header, footer and trailer content.
+   * 3. Create the request filename with the file path with respect to the institution code
+   * for the ECE request sent File.
+   * 4. Upload the content to the zoneB SFTP server.
+   * @param generatedDate date in which the eligible coe for
+   * particular institution is required.
+   * @returns Processing ECE request result.
+   */
+  async processECEFile(generatedDate?: string): Promise<ECEUploadResult[]> {
+    this.logger.log(`Retrieving eligible COEs for ECE request...`);
+    const eligibleCOEs = await this.disbursementScheduleService.getEligibleCOE(
+      generatedDate,
+    );
+    if (!eligibleCOEs.length) {
+      return [
+        {
+          generatedFile: "none",
+          uploadedRecords: 0,
+        },
+      ];
+    }
+    this.logger.log(`Found ${eligibleCOEs.length} COEs.`);
+    const fileRecords: Record<string, ECERecord[]> = {};
+    eligibleCOEs.forEach((eligibleCOE) => {
+      const institutionCode =
+        eligibleCOE.studentAssessment.offering.institutionLocation
+          .institutionCode;
+      if (!fileRecords[institutionCode]) {
+        fileRecords[institutionCode] = [];
+      }
+      fileRecords[institutionCode].push(this.createECERecord(eligibleCOE));
+    });
+    const uploadResult: ECEUploadResult[] = [];
+    try {
+      this.logger.log("Creating ECE request content...");
+      for (const institutionCode of Object.keys(fileRecords)) {
+        const eceUploadResult = await this.uploadECEContent(
+          institutionCode,
+          fileRecords,
+        );
+        uploadResult.push(eceUploadResult);
+      }
+    } catch (error) {
+      this.logger.error(
+        `Error while uploading content for ECE request: ${error}`,
+      );
+      throw error;
+    }
+    return uploadResult;
+  }
+
+  /**
+   * Upload the content in SFTP server location.
+   * @param institutionCode Institution code for the file generated.
+   * @param fileRecords Total records with institutionCode.
+   * @returns Updated records count with filepath.
+   */
+  async uploadECEContent(
+    institutionCode: string,
+    fileRecords: Record<string, ECERecord[]>,
+  ): Promise<ECEUploadResult> {
+    try {
+      // Create the Request content for the ECE request file by populating the content.
+      const fileContent = this.eceIntegrationService.createECEFileContent(
+        fileRecords[institutionCode],
```

Comment:
if its `fileRecords[institutionCode]`, then we can passed directly from the parent right